### ISSUE TITLE
Support VSE password for monitor script locations

### ISSUE THREAD

**Issue by lwatanabe2** [newrelic/terraform-provider-newrelic#1533]
### Feature Description
We have VSE enabled for our private location. I am hoping for the ability to specify the VSE password rather than the hmac in the location block for a monitor script, and for the hmac to be automatically computed and set for the monitor script in the location. This would make it so that we don’t need to precompute the hmacs for the scripts ourselves.

### Describe Alternatives
The alternative would be to precompute the hmacs for our scripts/locations and set it manually, but it would be much easier to have the hmacs computed and set automatically using the supplied VSE password.

### Additional context
example:
```
resource "newrelic_synthetics_monitor_script" "foo_script" {
  monitor_id = newrelic_synthetics_monitor.foo.id
  text = file("${path.module}/foo_script.js")
  location {
    name = "AWS_US_EAST_1"
    vse_password = "password"
  }
}
```

**Comment by zeffron** [comment]
This would make it possible to use VSE protected private locations with Synthetics monitoring scripts that use Terraform templating to include dynamic data from other resources. Precomputing the HMAC is not possible for this use case as the full text of the script is unknown until Terraform creates the other resources.

Additionally, ideally, the VSE password would be left out of the Terraform state and only the computed HMAC would be added to the state. This would prevent accidental leaking of the VSE password through the Terraform state, but allow Terraform to dynamically compute HMACs.

**Comment by mbazhlekova** [comment]
Support for VSE Password is available with version 2.34.0
